### PR TITLE
Feature/3

### LINKED ISSUE #3: Set Up A Basic Express Server
## Main Task
We will need to set up an Express server, content will be added later, we just need a framework to add content to.
The aim of the task is to set up a framework that points to different pages based on the requirements of the Pathfinder Task.

- [x] Set up a basic express server framework with a path for the root or homepage
- [x] Create a route for an individual phenomenon
- [x] Create a route for location list
- [x] Create a route for date / occurrence list

### Extras
- [x] Create a route for search page (optional)

### Documentation
[Express (Node.js)](https://expressjs.com/)

**Branch name:**
`feature/3#`

### PR DESCRIPTION
**Pull request by Minakrv**

### CODE REVIEW

File: src/db/connection.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+const connection = async () => {
+    try{
+
+    } catch (error) {
+        await mongoose.connect(process.env.MONGO_URI, {
+            useNewUrlParser: true,
+            useUnifiedTopology: true,
+            useFindAndModify : false
+        });
```
Comment (nmunro) [inline src/db/connection.js]: Why is this in the catch block, shouldn't it be in the try?

File: src/routes/about.route.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const express = require("express");
+const router = express.Router;
+
+router.get("/About", (req,res) => {
+    res.send("About Page");
```
Comment (nmunro) [inline src/routes/about.route.js]: Remember to use status codes, also don't use capital letters in route names, this should be "/about"

File: src/routes/location.route.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const express = require("express");
+const router = express.Router;
+
+const Axios = require("axios");
+
+router.get("/", (req,res) => {
+
```
Comment (nmunro) [inline src/routes/location.route.js]: This does nothing with the response and could hang your application, it doesn't have to do much, just a 200 status and a simple message is enough for now :)

File: src/routes/phenomenon.route.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const express = require("express");
+const app  = express();
+
+app.get("http://localhost:3000/phenomenon/" , (res, req) => {
+    app.post()
```
Comment (nmunro) [inline src/routes/phenomenon.route.js]: You can't post from the back end, you must modify the response object to have a status and either send a json message or some text.

File: src/routes/search.route.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+const express = require("express");
+const router = express.Router;
+
+
+// home page route
+router.get("/", (req,res) => {
+    res.send("Home Page");
```
Comment (nmunro) [inline src/routes/search.route.js]: Remember to use status codes :)

File: src/routes/search.route.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+});
+
+// about page route
+
+router.get("/about", (req, res) => {
+    res.send("About page");
```
Comment (nmunro) [inline src/routes/search.route.js]: Remember to use status codes :)

File: src/server.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
+const express = require("express");
+const app = express();
+const cors = require("cors");
+require("dotenv").config({ path: "./config.env" });
```
Comment (nmunro) [inline src/server.js]: Is there a reason you're changing the path to the env file?

File: src/server.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+const port = process.env.PORT || 3000;
+app.use(cors());
+app.use(express.json());
+
+
+
+//router
+app.get("./", (req,res)=> {
+    res.send("server");
```
Comment (nmunro) [inline src/server.js]: Remember to use status codes :)

File: src/server.js [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+const cors = require("cors");
+require("dotenv").config({ path: "./config.env" });
+const port = process.env.PORT || 3000;
+app.use(cors());
+app.use(express.json());
+
+
+
+//router
+app.get("./", (req,res)=> {
```
Comment (nmunro) [inline src/server.js]: Don't use '.' in route names, this is for a web server not your local file system :)

**Review by nmunro — CHANGES_REQUESTED** [review]

File: src/server.js (hunk 7 above)
Comment (RomanHyde) [reply src/server.js]: Would `require('dotenv').config();` be correct ?

File: src/routes/about.route.js (hunk 2 above)
Comment (RomanHyde) [reply src/routes/about.route.js]: like `res.status(200).send("About Page");` ?

Comment (nmunro) [reply src/routes/about.route.js]: Yup, that's exactly right :)

File: src/routes/phenomenon.route.js (hunk 4 above)
Comment (nmunro) [reply src/routes/phenomenon.route.js]: Also, remember to not hard code addresses like this, it should be "/phenomenon" with no trailing "/"

File: src/server.js (hunk 7 above)
Comment (nmunro) [reply src/server.js]: It depends, obviously I can't tell if there's a .env file, or if there's a config.env file that's ignored, so I'm a little unclear why this is different to how we've demonstrated things.

Comment (RomanHyde) [reply src/server.js]: I'll have to ask @Minakrv after lunch because I don't have access to her .env file.

**Comment by RomanHyde** [comment]
Hi @nmunro , I've fixed the changes you've requested and added some fixes of my own that I spotted as well. Could you have a look at the code when you are free? 

Also, it says this branch is 13 commits ahead and 9 commits behind the main branch, should I be worried about that?

File: src/routes/phenomenon.route.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+const express = require("express");
+const router = express.Router();
+const app = express();
```
Comment (nmunro) [inline src/routes/phenomenon.route.js]: Doesn't look like this line is needed anymore

File: src/routes/date.route.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+const express = require("express");
+const router = express.Router();
+const app = express();
```
Comment (nmunro) [inline src/routes/date.route.js]: Doesn't look like this is used.

File: src/routes/phenomenon.route.js [hunk 12]
```diff
@@ -0,0 +1,9 @@
...
+
+router.get("/", (res, req) => {
+    res.status(200).send("Phenomenon page");
+        });
```
Comment (nmunro) [inline src/routes/phenomenon.route.js]: Mind your indentation :P

**Review by nmunro — DISMISSED** [review]

File: src/routes/phenomenon.route.js (hunk 10 above)
Comment (RomanHyde) [reply src/routes/phenomenon.route.js]: Oooh noted! I might remove that.

File: src/routes/phenomenon.route.js (hunk 12 above)
Comment (RomanHyde) [reply src/routes/phenomenon.route.js]: Yes, ewww. I need to fix that. Thank you!

**Review by nmunro — APPROVED** [review]

**Pull request merged**